tests/kernel/gen_isr_table/src/main.c

```diff
@@ -100,14 +100,14 @@ ISR_DIRECT_DECLARE(isr2)
 }
 #endif
 
-void isr3(const void *param)
+void isr3(const struct device *param)
```

Comment:
For sure I'm missing something here. But why all the ISR parameters must necessarily be a `struct device`?

Reply:
The parameter passed to the ISR is defined in `IRQ_CONNECT` macro [Defining a regular ISR](https://docs.zephyrproject.org/latest/reference/kernel/other/interrupts.html#defining-a-regular-isr). Most of the time drivers pass `struct device` but in a few cases that you pointed out that doesn't seem to be a case so it indeed should be fixed.

Reply:
and the script I run did not check that the actual parameter is being used an actual struct device then. Now it's fixed.